Triangulation_2/examples/Triangulation_2/polygon_triangulation.cpp

```diff
@@ -2,22 +2,16 @@
 #include <CGAL/Constrained_Delaunay_triangulation_2.h>
 #include <CGAL/Triangulation_face_base_with_info_2.h>
 #include <CGAL/Polygon_2.h>
-#include <iostream>
-
-struct FaceInfo2
-{
-  FaceInfo2(){}
-  int nesting_level;
+#include <CGAL/mark_domains.h>
```

Comment:
The name of the header seems a bit generic

Reply:
Another possibility would be to add it as member function to the `Constrained_triangulation_2` class.

Reply:
I prefer @afabri 's suggestion but alternatively, it could be `CGAL/constrained_triangulation_mark_domains.h`

Reply:
The name needs probably some discussion. Should there be a `_2` somewhere?   Is it domain or domains, the former having potentially several connected components?  etc.

Reply:
What actually is a domain geometrically speaking? As far as I understood this is a way of finding holes, but then there could be an island inside the hole. There must be a geometric term for islands within islands. (At least I would have thought so with geo meaning earth in latin :laughing: )

Reply:
For the Delaunay mesh generator the domain is the area where the refinement shall happen.  It can be controlled by seed points, which can be placed in holes as well as in non-holes.   The function I wrote considers the unbounded area as outside and then alternates between inside and outside the domain.  That's why we speak about nesting level and parity.

Reply:
See  [here](https://github.com/CGAL/cgal/pull/6412/files#diff-8a166ef6d855401ed695deab613fefe7164c2832a17f3fa5e104f0bd42503719R8)

Reply:
It seems like `domains`, `bounded domains`, `regions`, `bounded regions` or even `islands` (used in https://doc.cgal.org/latest/Polyline_simplification_2/index.html#Chapter_2D_Polyline_simplification) would be acceptable terms. I am not sure I understand how a non-plural term such as `domain` would apply when there can be more than one of them.

Reply:
Oh, would you say the `island` inside a lake inside an `island`, all belong to the same domain? In that case, I understand why you would use a non-plural.

Reply:
In geography, the lithosphere is a domain concerning land. If that makes for a good analogy?

Reply:
Reading [this ](https://en.wikipedia.org/wiki/Domain_(mathematical_analysis)) wikipedia page we probably should use plural.

Reply:
But I will stick to singular, because in the [User Manual](https://doc.cgal.org/latest/Mesh_2/index.html#title5) we define domain as something that can have several connected components.

Reply:
Maybe the best way to explain it in the documentation would be an image, for example, the orange coloured parts belong to the same domain:
![image](https://user-images.githubusercontent.com/444143/158552151-39a153ce-eb66-4e7a-8a19-c5eca4383444.png)

Reply:
Oh like this image:
![image](https://user-images.githubusercontent.com/444143/158554955-6887bf9f-a375-4299-92ee-16af6f4b8408.png)

Reply:
It seems a bit odd to say that the green triangle is "connected" to the other green parts, but ok.